### ISSUE TITLE
[bqplot 1.0 release] discussion

### ISSUE THREAD

**Issue by ibdafna** [bqplot/bqplot#1264]
Using this issue to track 1.0 release topics brought up during the weekly meetings. I will update this note with more items as we continue the discussion.

- Implement missing `pyplot` interface methods. Consider potential renaming of the pyplot interface with 1.0
- Refactor front-end code to use camelCase. Linting with Prettier?
- Merge both ScatterGL/Scatter into a single Scatter mark. Remove Three.js dependency in favour of canvas rendering?

**Comment by martinRenou** [comment]
- bqscales: https://github.com/bqplot/bqplot/pull/904

**Comment by ChakriCherukuri** [comment]
I want the following items to be added to 1.0:

1. Fix pyplot issues for basic marks
2. Rename MarketMap to TileMap
3. Add the following compound marks:
    - radar
    - date range selector
    - fishbone
    - triangle

**Comment by martinRenou** [comment]
- Drop ThreeJS dependency (@maartenbreddels). It is a pretty huge dependency for only one mark. Would we still have decent performances if replaced by Web2DCanvas API? Another idea could be to use WebGL directly, our usage of ThreeJS is low-level enough that using WebGL directly might not add too much complexity.

**Comment by acelere** [comment]
When we rename MarketMap to TileMap, I would like to keep a wrapper as MarketMap and issue a "soon to be deprecated" message for one or two releases in order to allow for a smooth transition. Then delete the wrapper.

**Comment by maartenbreddels** [comment]
I don't think we want to go to canvas, that's gonna be orders of magnitude slower, and for extensions such as bqplot-image-gl to work, we need to provide a WebGL context. Getting rid of threejs might be possible, but it will take some work.

What exactly is the issue with threejs? I don't think the bqplot package is that large, is it maybe loading the bqplot package from Jupyter Lab taking too long? If that is the issue,  think an easier solution would be to lazily load threejs, similar to what @bollwyvl has done in https://github.com/bqplot/bqplot/pull/1192

**Comment by martinRenou** [comment]
> What exactly is the issue with threejs?

I wouldn't say there is an issue with ThreeJS itself, apart from its size maybe. But our usage of ThreeJS in bqplot is very limited, and so low-level that I start to think that using WebGL directly would be enough.
